builds/jazz-build-module/lambda-events-module.groovy

```diff
@@ -0,0 +1,175 @@
+      def existing_lambda_configs = existing_notifications.LambdaFunctionConfigurations
+      new_events = getLambdaEvents(existing_notifications, events)
+      if (new_events != null && new_events.size() > 0) {
+        for (item in existing_lambda_configs) {
+          lambdaFunctionConfigurations.add(item)
+        }
+        new_lambda_configuration.Events = new_events
+        lambdaFunctionConfigurations.add(new_lambda_configuration)
+      }
+    } else {
+      new_events = checkAndConvertEvents(events)
+      new_lambda_configuration.Events = new_events
+      lambdaFunctionConfigurations.add(new_lambda_configuration)
+    }
+    if (lambdaFunctionConfigurations != null && lambdaFunctionConfigurations.size() > 0) {
+      existing_notifications.LambdaFunctionConfigurations = lambdaFunctionConfigurations
+      def newNotificationJson = JsonOutput.toJson(existing_notifications)
+      def response = sh(returnStdout: true, script: "aws s3api put-bucket-notification-configuration --bucket $s3BucketName --notification-configuration \'${newNotificationJson}\' --output json")
+    }
+  } catch (ex) {
+  }
+}
+
+def getLambdaEvents(existing_notifications, events){
+  def existing_events = []
+  def new_events = []
+  for (item in events) {
+    new_events.add(item)
+  }
+  for (item in existing_notifications.LambdaFunctionConfigurations) {
+    existing_events.addAll(item.Events)
+  }
+  def cleanupIndex = -1
+  echo "events . $events"
+
+  // Removing the existing events from the new event list
+  def event_names
+  for (item in events) {
+    cleanupIndex++
+    if ((item.contains("ObjectCreated") || item.contains("ObjectRemoved")) &&
+      (existing_events.contains("s3:ObjectCreated:*") || existing_events.contains("s3:ObjectRemoved:*"))) {
+      new_events[cleanupIndex] = null
+    } else if (existing_events.contains(item)) {
+      new_events[cleanupIndex] = null
+    }
+  }
+  new_events.removeAll([null])
+  def events_list = []
+  if (new_events.size() > 0 && new_events != null) {
+    events_list = checkAndConvertEvents(new_events)
+  }
+  return events_list
+}
+
+def checkAndConvertEvents(events){
+  def new_events = []
+  // converting the new events to * event
+  def cleanupIndex = -1
+  def isCreationEvent = false
+  def isRemovalEvent = false
+  if (events.size() > 0 && events != null) {
+    for (item in events) {
+      new_events.add(item)
+    }
+
+    for (item in events) {
+      cleanupIndex++
+      if (item.contains("ObjectCreated")) {
+        isCreationEvent = true
+        new_events[cleanupIndex] = null
+      } else if (item.contains("ObjectRemoved")) {
+        isRemovalEvent = true
+        new_events[cleanupIndex] = null
+      }
+    }
+
+    new_events.removeAll([null])
+    if (isCreationEvent == true) {
+      new_events.add("s3:ObjectCreated:*")
+    }
+    if (isRemovalEvent == true) {
+      new_events.add("s3:ObjectRemoved:*")
```

Comment:
This should not be * by default. If user selected ObjectRemoved:DELETE or something else then we should respect that choice. * will only be used in this(ObjectRemoved) case only if user had selected that to begin with. 

You can use * only for ObjectCreated by default.

Reply:
done